File: ExamenUnidad2__JAVA/ExamenUnidad2/src/examenunidad2/Visita.java

package examenunidad2;

import java.util.ArrayList;
import java.util.Scanner;

public class Visita 
{
    Scanner leer = new Scanner (System.in);

    public static int IDENTIFICADOR = 90;
    private int guiaCargo;
    double costoTotal;
    private int cantidadNiños;
    private int cantidadAdultos;
    String fechaDeLaVisita;
    int idVisita ;
    ArrayList<Visitantes> listaDeVisitantes = new ArrayList<Visitantes>();
    
    public Visita() {
    }

    public Visita(int guiaACargo, String fechaDeLaVisita, 
                  int cantidadAdultos, int cantidadNiños) 
    {
        this.guiaCargo = guiaCargo;
        this.fechaDeLaVisita = fechaDeLaVisita;
        this.idVisita = IDENTIFICADOR;
        this.cantidadAdultos = cantidadAdultos;
        this.cantidadNiños = cantidadNiños;
        IDENTIFICADOR++;
    }

    public double getCostoTotal() {
        return costoTotal;
    }

    public int getCantidadNiños() {
        return cantidadNiños;
    }

    public int getCantidadAdultos() {
        return cantidadAdultos;
    }

    public String getFechaDeLaVisita() {
        return fechaDeLaVisita;
    }
    
    public int getIdVisita() {
        return idVisita;
    }

    public static int getIDENTIFICADOR() {
        return IDENTIFICADOR;
    }

    public static void setIDENTIFICADOR(int IDENTIFICADOR) {
        Visita.IDENTIFICADOR = IDENTIFICADOR;
    }

    public int getGuiaCargo() {
        return guiaCargo;
    }

    public void setGuiaCargo(int guiaCargo) {
        this.guiaCargo = guiaCargo;
    }

    public ArrayList<Visitantes> getListaDeVisitantes() {
        return listaDeVisitantes;
    }

    public void setListaDeVisitantes(ArrayList<Visitantes> listaDeVisitantes) {
        this.listaDeVisitantes = listaDeVisitantes;
    }
    
    
    public double calcularCostoTotal(ArrayList<Visitantes> visitantes) 
    {
        double costoAdulto = 100;
        double costoNino = 50;
        int totalAdultos = 0;
        int totalNinos = 0;
        
        for (Visitantes visitante : visitantes)
        {
            if (visitante.calcularEdad() >= 18) 
            {
                totalAdultos++;
            } 
            else 
            {
                totalNinos++;
            }
        }
 
        
        double costoTotal = (totalAdultos * costoAdulto) + (totalNinos * costoNino);
        double descuento = 1 - visitantes.stream().mapToDouble(Visitantes::obtenerDescuento).sum();
        double costoFinal = costoTotal * descuento;
        return costoFinal;
    }

    
    
    
}

File: ExamenUnidad2_PYTHON 1/ExamenUnidad2_PYTHON 1/ExamenUnidad2/Visita.py
from Visitantes import Visitantes

class Visita:
    IDENTIFICADOR = 90
    guia_cargo = None
    costo_total = None
    cantidad_niños = 0
    cantidad_adultos = 0
    fecha_de_la_visita = None
    id_visita = 0
    lista_de_visitantes = []

    def __init__(self, id_guia=None, fecha_de_la_visita=None, cantidad_adultos=None, cantidad_niños=None):
        if id_guia is not None and fecha_de_la_visita is not None and cantidad_adultos is not None and cantidad_niños is not None:
            self.id_guia = id_guia
            self.fecha_de_la_visita = fecha_de_la_visita
            self.cantidad_adultos = cantidad_adultos
            self.cantidad_niños = cantidad_niños
        else:
            # Aquí maneja el caso donde no se proporcionan todos los argumentos
            pass
    

    
    def calcular_costo_total(self, lista_de_visitantes = []):
        costo_adulto = 100
        costo_nino = 50
        total_adultos = 0
        total_ninos = 0

        for visitante in lista_de_visitantes:
            if visitante.calcular_edad() >= 18:
                 total_adultos += 1
            else:
                total_ninos += 1
                
        costo_total = (total_adultos * costo_adulto) + (total_ninos * costo_nino)

        descuento = 1 - sum(visitante.obtener_descuento() for visitante in lista_de_visitantes)
        costo_final = costo_total * descuento
        print(descuento)
        return costo_final

    def agregar_visitante(self, visitante):
        self.lista_de_visitantes.append(visitante)

    def get_guia_cargo(self):
        return self.guia_cargo

    def set_guia_cargo(self, guia_cargo):
        self.guia_cargo = guia_cargo

    def get_fecha_de_la_visita(self):
        return self.fecha_de_la_visita

    def get_id_visita(self):
        return self.id_visita

    def get_cantidad_adultos(self):
        return self.cantidad_adultos

    def get_cantidad_niños(self):
        return self.cantidad_niños
